src/utils/Logger.ts

```diff
@@ -182,13 +256,28 @@ class Logger {
       warn: console.info,
     }
 
-    const writeLog = (level: string, message: string) => {
-      const timestamp = new Date().toISOString()
-      RNFS.appendFile(logFilePath, `${level} [${timestamp}] ${message}\n`, 'utf8').catch(
-        (error) => {
-          consoleFns.debug(`Failed to write to ${logFilePath}`, error)
+    const writeLog = async (level: string, message: string) => {
+      try {
+        // If log folder not present create it
+        if (!(await RNFS.exists(this.getReactNativeLogsDir()))) {
+          await RNFS.mkdir(this.getReactNativeLogsDir())
+        }
+
+        // If daily log file is not present create it
+        if (!(await RNFS.exists(logFilePath))) {
+          await RNFS.writeFile(logFilePath, '', 'utf8')
         }
-      )
+
+        const timestamp = new Date().toISOString()
+        // Ensure messages are converted to utf8 as some remote CTA's can have non utf8 characters
```

Comment:
kinda a nit: i'm not sure i understand this comment.  do you mean CTAs have non-ascii characters (i.e., unicode ones), so we must encode using utf-8?

Reply:
Can you fix the comment?

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/2680/commits/82a3162bcb042760961e008c0132769d04f6ba48!